security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -646,7 +662,14 @@ func (sc *SecretCache) rotate(updateRootFlag bool) {
 				// If token is still valid, re-generated the secret and push change to proxy.
 				// Most likey this code path may not necessary, since TTL of cert is much longer than token.
 				// When cert has expired, we could make it simple by assuming token has already expired.
-				ns, err := sc.generateSecret(context.Background(), secret.Token, connKey, now)
+				var ns *model.SecretItem
+				var err error
+				if UseRotationWithCert {
```

Comment:
how does this work today when using PROV_CERT?

Reply:
And why the 'if' ? !UserRotationWithCert as last param ?

Reply:
UseRotationWithCert is only for rotate cert function. So when the user choose to use rotate with cert we will use the stored old valid cert in OUT_CERTS Path. PROV_CERT is only used for the client first start. 

So I fully understand costin's concern: the new added field will affect the existing the process. However in this PR I am adding a **NEW** Process which are fully **isolated** from the existing the workflow.

The new workflow: only when cert needs to rotate we add the client connection reconnect, and this kind of process is marked with a isRotated(True). This can guarantee that it will not be confused with the existing workflow

The Existing workflow: 1. connection first start 2. connection is restarted
for both the case the connection is built with the cert provided in the PROV_CERT file path if PROV_CERT is not empty. which is the default behavior. And isRotated is set to False